crates/rpc/api/src/lib.rs

```diff
@@ -1,18 +1,32 @@
 //! Macro trait def for the `strata_` RPC namespace using jsonrpsee.
 use bitcoin::Txid;
 use jsonrpsee::{core::RpcResult, proc_macros::rpc};
-use strata_db::types::L1TxStatus;
+use strata_db::types::{L1TxEntry, L1TxStatus};
 use strata_primitives::bridge::{OperatorIdx, PublickeyTable};
 use strata_rpc_types::{
     types::{RpcBlockHeader, RpcClientStatus, RpcL1Status},
-    HexBytes, HexBytes32, L2BlockStatus, RpcBridgeDuties, RpcCheckpointInfo, RpcDepositEntry,
-    RpcExecUpdate, RpcSyncStatus,
+    HexBytes, HexBytes32, L2BlockStatus, RpcBridgeDuties, RpcChainState, RpcCheckpointInfo,
+    RpcDepositEntry, RpcExecUpdate, RpcSyncStatus,
+};
+use strata_state::{
+    block::L2Block, client_state::ClientState, id::L2BlockId, operation::ClientUpdateOutput,
+    sync_event::SyncEvent,
 };
-use strata_state::{id::L2BlockId, operation::ClientUpdateOutput, sync_event::SyncEvent};
-
 #[cfg_attr(not(feature = "client"), rpc(server, namespace = "strata"))]
 #[cfg_attr(feature = "client", rpc(server, client, namespace = "strata"))]
 pub trait StrataApi {
+    /// Get blocks at a certain height
+    #[method(name = "getBlocksAtIdx")]
+    async fn get_blocks_at_idx(&self, idx: u64) -> RpcResult<Vec<HexBytes32>>;
+
+    /// Get the ChainState at a certain index
+    #[method(name = "getChainstateAtIdx")]
+    async fn get_chainstate_at_idx(&self, idx: u64) -> RpcResult<Option<RpcChainState>>;
+
+    /// Get the ClientState at a certain index
+    #[method(name = "getClientStateAtIdx")]
+    async fn get_clientstate_at_idx(&self, idx: u64) -> RpcResult<Option<ClientState>>;
```

Comment:
Should these also be a part of debug RPC?

Also, would it make sense to have a new type `RpcClientState` instead of exposing the `ClientState` if this is not a part of debug RPC?

Reply:
I think it does make sense to have them in debug_. The notion behind types Rpc* is due to Serialization trait not being derived for those structs or to have a stripped down version of the struct. If we move these to debug then this should not be an issue.
cc: @delbonis